### PR TITLE
[KNative Serving] adding Helm 2 annotations to CRDs

### PR DESCRIPTION
**Pull request by akirillov**
**What type of PR is this?**
bug

**What this PR does/ why we need it**:
This PR is a follow-up to https://github.com/mesosphere/charts/pull/834. It adds Helm webhook annotations back to CRDs for Helm 2. The annotations were mistakenly removed in the previous PR.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

**Checklist**

* [x] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/knative/charts/serving/crds/crds.yaml [hunk 1]
```diff
@@ -26,7 +27,6 @@ kind: CustomResourceDefinition
 metadata:
   name: certificates.networking.internal.knative.dev
   labels:
-    serving.knative.dev/release: {{ .Chart.AppVersion | quote }}
```
Comment (armandgrillet) [inline staging/knative/charts/serving/crds/crds.yaml:29]: What is the reason behind removing this label?

**Review by armandgrillet — APPROVED** [review]
LGTM.

File: staging/knative/charts/serving/crds/crds.yaml (hunk 1 above)
Comment (akirillov) [reply staging/knative/charts/serving/crds/crds.yaml:29]: I was getting the following error during the CRD installation steep with Helm 3:
```
Error: failed to install CRD crds/crds.yaml: error parsing : error converting YAML to JSON: yaml: invalid map key: map[interface {}]interface {}{".Chart.AppVersion | quote":interface {}(nil)}
```
According to [Helm docs](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-1-let-helm-do-it-for-you):
> There is now a special directory called crds that you can create in your chart to hold your CRDs. These CRDs are not templated, but will be installed by default when running a helm install for the chart.

As these labels are not used anywhere in KNative code, I've just removed them to resolve the templating issue.

**Comment by akirillov** [comment]
soft ping @alejandroEsc and @joejulian. Can you please provide you reviews as you're familiar with the previous PR for KNative Helm 3 support? Thanks!

File: staging/knative/charts/serving/files/crds.yaml [hunk 2]
```diff
@@ -395,9 +414,11 @@ spec:
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
+  annotations:
+    helm.sh/hook: crd-install
+    helm.sh/hook-delete-policy: before-hook-creation
```
Comment (alejandroEsc) [inline staging/knative/charts/serving/files/crds.yaml:419]: `helm.sh/hook-delete-policy: before-hook-creation` these are not recommended. These will delete any resources whenever they are run that are of the types defined by the CRD. I would remove these.

**Review by alejandroEsc — COMMENTED** [review]
One item of concern, but lgtm overall.

File: staging/knative/charts/serving/files/crds.yaml (hunk 2 above)
Comment (akirillov) [reply staging/knative/charts/serving/files/crds.yaml]: Thanks for pointing that out. Removed.

**Review by joejulian — APPROVED** [review]

**Pull request merged**